protostar/commands/test/environments/test_execution_environment.py

```diff
@@ -52,7 +56,8 @@ async def invoke(self, function_name: str) -> Optional[ExecutionResourcesSummary
 
         async with self._expect_revert_context.test():
             async with self._finish_hook.run_after():
-                tx_info = await self.perform_invoke(function_name)
+                with redirect_stdout(self.state.output_recorder.record(output_name)):
```

Comment:
this might be nice to be encapsulated as:

```suggestion
                with self.state.output_recorder.redirect(output_name):
```

1. you can use `@contextmanager` decorator here
2. I have deliberately removed the `_stdout` suffix from the name, in order to make an API that will be agnostic to recording stdout/stderr.